agent/semgrep_agent.py

```diff
@@ -94,6 +95,12 @@ def _run_analysis(
     return (output.stdout, output.stderr)
 
 
+def _is_text_file(content: str | bytes) -> bool:
```

Comment:
```suggestion
def _is_printable_content(content: str | bytes) -> bool:
```
When I read this function name, I suppose that it will get a file as a parameter and check either its content, path, or other related data.

I would change the name to _is_printable_text or _is_printable_content to reflect what the function actually does.